src/Explorer/Menus/NotificationConsole/NotificationConsoleComponent.tsx

```diff
@@ -37,15 +37,15 @@ export interface ConsoleData {
 
 export interface NotificationConsoleComponentProps {
   isConsoleExpanded: boolean;
-  onConsoleExpandedChange: (isExpanded: boolean) => void;
-  consoleData: ConsoleData[];
-  onConsoleDataChange: (consoleData: ConsoleData[]) => void;
+  consoleData: ConsoleData;
+  inProgressMessageIdToBeDeleted: string;
```

Comment:
Is `inProgressMessageIdToBeDeleted` needed anymore? Now that we set `consoleData` directly can it be removed? Seems weird to have both. Feel free to also say this is out of scope.

Reply:
The tricky thing about `consoleData` is that we are just adding it to the list of `consoleData` that were added previously. Before my change, we used to maintain a knockout list of all console data that are displayed in the console in `Explorer.ts`. I wanted to remove this dependency so I moved the list inside the component itself. Now Explorer.ts is only responsible for telling the console what <b>NEW</b> console data to display. However, this also means that we won't be able to delete a specific console data directly in `Explorer.ts`. We have to pass the message id to the component so that we can remove it from the list.

Reply:
Let me know what you think about this. Another problem with maintaining the list inside Explorer is the clear all notifications button since we can't just clear them within the component. The component would have to tell Explorer to clear all the notifications which creates a unnecessary coupling.

Reply:
Got it. Yeah I am ok with leaving it then. Maybe put a //TODO to refactor it once we move everything to KO.